src/components/calcite-switch/calcite-switch.tsx

```diff
@@ -60,109 +141,85 @@ export class CalciteSwitch {
     }
   }
 
-  @Listen("click") onClick(e: MouseEvent): void {
+  @Listen("click")
+  onClick(e: MouseEvent): void {
     // prevent duplicate click events that occur
     // when the component is wrapped in a label and checkbox is clicked
     if (
-      (!this.disabled && this.el.closest("label") && e.target === this.inputProxy) ||
+      (!this.disabled && this.el.closest("label") && e.target === this.inputEl) ||
       (!this.el.closest("label") && e.target === this.el)
     ) {
-      this.updateSwitch(e);
+      this.toggle();
     }
   }
 
-  @Listen("keydown") keyDownHandler(e: KeyboardEvent): void {
+  @Listen("keydown")
+  keyDownHandler(e: KeyboardEvent): void {
     const key = getKey(e.key);
     if (!this.disabled && (key === " " || key === "Enter")) {
-      this.updateSwitch(e);
+      this.toggle();
     }
   }
 
-  @Watch("switched") switchWatcher(): void {
-    this.switched
-      ? this.inputProxy.setAttribute("checked", "")
-      : this.inputProxy.removeAttribute("checked");
-  }
-
-  private inputProxy: HTMLInputElement;
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
 
   connectedCallback(): void {
-    this.setupProxyInput();
+    this.guid = this.el.id || `calcite-switch-${guid()}`;
   }
 
-  disconnectedCallback(): void {
-    this.observer.disconnect();
+  componentDidLoad(): void {
+    this.renderInput();
   }
 
-  componentWillRender(): void {
-    this.syncProxyInputToThis();
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  private renderInput(): void {
+    this.switched && this.inputEl.setAttribute("checked", "");
+    this.inputEl.disabled = this.disabled;
+    this.inputEl.id = `${this.guid}-input`;
+    this.inputEl.name = this.name;
+    this.inputEl.style.setProperty("bottom", "0", "important");
```

Comment:
nitpick: Might be nice to just set the whole css incase there are any other styles that might be problematic?

```
.style.cssText = `
bottom: 0 !important;
...
`
```

Do you think this would be useful for other components as a utility?

Reply:
It would be a nice utility, definitely.  i'm using this same thing in at least two other components, radio and checkbox.